### ISSUE TITLE
Have servo build on Windows without the need of msys/mingw

### ISSUE THREAD

**Issue by Jayflux** [servo/servo#10893]
It would be great to simplify the build steps of servo on Windows platforms.
Currently there is a lot of work to do, with setting up mingw, python and msys.

It would be ideal to just be able to use powershell or a bash terminal within a windows environment and have it build.

This obviously means we need to remove linux-only dependancies such as make

**why?**
It would encourage more contributors who use windows systems if the setup steps were less combersome, and not having to set up a linux system within a windows OS

**What needs to be done?**
- We could start with an Audit of what we're using from msys and how we can migrate to other alternatives (for example make)
- Rewrite some tools in a cross-platform way, or use alternatives

This will need some input from @larsbergstrom @nox  et al

**Comment by Ms2ger** [comment]
Not much point in doing anything for Servo itself before mozilla-central moves off autoconf and make; that's probably a few years off.

There might be other dependencies that use make; possibly it would be useful for the wider Rust community if those were to avoid make.

**Comment by zeusintuivo** [comment]
I have researched and documented an effective way to have windows environments set up for 32 and 64 bits in Windows 7. I assume this will work in all windowses. 
Please see documentation I posted in this ticket. #12356

**Comment by Jayflux** [comment]
@zeusintuivo do you actually need Visual Studio now?
I remember Microsoft doing some work in their SDK so you could get Visual C++ and all the gubbins without having download Visual Studio.

They released Visual C++ build tools (install using the Default Install option)
http://landinghub.visualstudio.com/visual-cpp-build-tools

**Comment by Jayflux** [comment]
```
Install Community 2015 Visual Studio - https://www.visualstudio.com/en-us/downloads/download-visual-studio-vs.aspx#d-community - Like 3 hours
+ Direct Link Download - https://go.microsoft.com/fwlink/?LinkId=691978
+ About 15 Gigas to 35 Gigas. Depends on how much you install.

Install MSYS2 - https://msys2.github.io/  Like 30 minutes
+ Direct Link Download - http://repo.msys2.org/distrib/x86_64/msys2-x86_64-20160205.exe
+ About 3 GBs

Install cmder - http://cmder.net/  Like 20 minutes
+ Direct Link Download - https://github.com/cmderdev/cmder/releases/download/v1.3.0/cmder.zip
+ Download full editor - 80 MB
+ Create "_" folder: C:\_   
+ unzip under folder  /_
+ click cmder ( select unblock all)
+ Configure cmder: 
----+ Select 'Settings'
----* Integration-> Default Term ...Check  Register OS on Startup
----* Main->Size & Pos  ...Check Autosave window size and position
----* Main->Size & Pos  ...Check Restore active monitor




Install Path Editor - http://www.rapidee.com/en/download
+ Direct Link Download - http://www.rapidee.com/download/RapidEE_setup.exe
+ Install and open 

Install Choco - https://chocolatey.org/instal   Like 
+ Start cmder  - as ADMIN (by right clicking it)
---+ Create tools folder
-------: mkdir C:\tools
---+ Paste this:
--------:  @powershell -NoProfile -ExecutionPolicy Bypass -Command "iex ((new-object net.webclient).DownloadString('https://chocolatey.org/install.ps1'))" && SET PATH=%PATH%;%ALLUSERSPROFILE%\chocolatey\bin

CHECK Choco Paths:
+ Open Rapidee Program, if it was open refresh (F5)
+ Should look something like this:
--+ Values:
-------------: ChocolateyInstall=C:\ProgramData\chocolatey
-------------: ChocolateyLastPathUpdate=Fri Jul 15 14:56:57 2016
-------------: ChocolateyToolsLocation=C:\tools
## Remember to Click "SAVE" when to apply changes ^. 


Install PYTHON 
+ RE-Start cmder  - as ADMIN (by right clicking it)
+ Using choco install python 2.7.11 type:
-----: choco install -y python2 

CHECK PYTHON PATHS :
+ Open 'Rapidee' Program, if it was open click refresh (F5)
--+ Values 3 paths inside PATH:
-------------: Add Inside to the bottom of the list PATH: .... C:\tools\python2:C:\tools\python2\Lib:C:\tools\python2\Scripts:
--+ Values Add 2 more Variables:
-------------: PYTHONHOME=C:\tools\python2
-------------: PYTHONPATH=C:\tools\python2\Lib
## Remember to Click "SAVE" when to apply changes ^.

Test Python:
+ RE-Start cmder  - as ADMIN (by right clicking it)
+ Type to enter python console:
-------------: python -v
######## If this ^ does not take you to python interactive shell. Then python or ENVIRONMENT PATHs are not SET CORRECTLY; do not continue until you get this right. 


Install PYTHON  PIP VIRTUALENV 
+ RE-Start cmder  - as ADMIN (by right clicking it)
+ Using choco install pip
-------------: choco install -y pip
-------------: python -m pip install --upgrade pip
-------------: pip install virtualenv
-------------: python --version


Add PACMAN to PATHS:
+ Open 'Rapidee' Program, if it was open click refresh (F5)
--+  Add as the first element to PATH:
-------------: C:\msys64\usr\bin
## Remember to Click "SAVE" when to apply changes ^.


Install Rust Requirements - https://github.com/rust-lang/rust  - Like 1 hour
+ Open CMDER as Admin
---+ Go to: C:\msys32\usr\bin
--------------: cd C:\msys64\usr\bin
---+Execute:
--------------:  pacman -Sy pacman-mirrors
--------------:  pacman -S make diffutils mingw-w64-x86_64-cmake mingw-w64-x86_64-gcc
--------------:  pacman -S mingw-w64-x86_64-gdb  make  mingw-w64-x86_64-cmake  mingw-w64-x86_64-pkg-config mingw-w64-x86_64-harfbuzz mingw-w64-x86_64-cairo mingw-w64-x86_64-fontconfig mingw-w64-x86_64-glib2 mingw-w64-x86_64-icu  mingw-w64-x86_64-freetype  mingw-w64-x86_64-physfs mingw-w64-x86_64-libvorbis  mingw-w64-x86_64-flac  mingw-w64-x86_64-dumb  mingw-w64-x86_64-libtheora  mingw-w64-x86_64-libjpeg-turbo

Upgrade PYTHON SETUPTOOLS - https://pypi.python.org/pypi/setuptools/18.5#downloads - Like 20 minutes
+ Direct Link Download - https://bootstrap.pypa.io/ez_setup.py
+ Go to download location:
------------: cd C:\Users\%USERNAME%\Downloads
+ Execute Downloaded file:
------------: python ez_setup.py
######## If there red errors ^ running this running it again.

INSTALL CMAKE For Windows  - https://cmake.org/download/
+ Direct Link Download - https://cmake.org/files/v3.6/cmake-3.6.0-win64-x64.msi
+ Make sure the CMAKE Directory is the last one on the list for tht %PATH% VAR.

BUILD Servo
+ cd C:/_/
+ git clone --depth 1 --single-branch https://github.com/servo/servo
+ cd servo
----: sh mach build --dev
----: sh mach run --dev tests/html/about-mozilla.html

DEBUG BUILD
----:  set RUST_BACKTRACE=1 & sh mach build --dev --verbose
```

**Comment by Coder206** [comment]
@Jayflux thanks! I was curious to know if it was possible.

**Comment by zeusintuivo** [comment]
@Jayflux Thanks, About the  Visual C++ build tools, I did not know about those. But if you can try it and if it works. Then you could update the Post. Notice, I made two virtual machines with both 32 and 64 bits. I tried several things and even based on Mozilla's building requirements Visual Studio 2015 was recommended. I hate installing such a huge amount of data as well by installing VB. 

I have come up with a stable way to do it. And I was able to do it twice on brand new installations.  My goal was accomplished by providing a way for Ubuntu and Windows to install. I called it "draft," if anyone wants to refactor and optimize the results of my research. You are all welcome to hack into it and provides us with an updated version.

**Comment by zeusintuivo** [comment]
To answer your question: "Have servo build on Windows without the need of msys/mingw" Right now the answer is: NO. you cannot. Someone will have to change the code to start using Microsoft libraries instead of mingw's.

**Comment by zeusintuivo** [comment]
@Jayflux also notice that you only posted the 64bit solution, the libraries for 32bits, are named i686

**Comment by Jayflux** [comment]
> To answer your question: "Have servo build on Windows without the need of msys/mingw" Right now the answer is: NO. you cannot. Someone will have to change the code to start using Microsoft libraries instead of mingw's.

Which code is this?
Do you have any examples?

**Comment by vvuk** [comment]
We're switching almost all dependencies to cmake in issue #11756 -- this will remove most of the dependencies on msys, because we'll use cl and msbuild to build most binary pieces.  servo/components/script will require the ninja build tool (just a single .exe download) due to parallelism issues in msbuild, still built with cl.

However, spidermonkey (mozjs) will continue to require a number of the msys tools for the build, as that build system is custom and requires them.  That's going to be the blocking thing to not require any of msys etc.

*However*, if we can hold our noses, we can always have golden builds of mozjs get downloaded for people who don't have the build tools installed, since it doesn't change all that often...

**Comment by cbrewster** [comment]
Fixed by #11756
